aea/agent.py

```diff
@@ -199,14 +199,20 @@ def start(self) -> None:
 
         :return: None
         """
+        self._start_setup()
+        self._run_main_loop()
+
+    def _start_setup(self) -> None:
```

Comment:
Great idea! Please make sure the doc string below is clean (correct indentation)